core/filter_api.php

```diff
@@ -1063,6 +1065,17 @@ function filter_get_query_sort_data( array &$p_filter, $p_show_sticky, array $p_
 	if( gpc_string_to_bool( $p_filter[FILTER_PROPERTY_STICKY] ) && ( null !== $p_show_sticky ) ) {
 		$p_query_clauses['order'][] = '{bug}.sticky DESC';
 	}
+	
+	# For user fields, the complete definition of the sort_field is deferred
+	# until for loop below
+	if( !isset( $s_fk_map ) ) {
+		$s_fk_map = array(
+				'category_id' => array( 'table' => '{category}', 'sort_field' => 'name' ),
+				'project_id'  => array( 'table' => '{project}' , 'sort_field' => 'name' ),
+				'handler_id'  => array( 'table' => '{user}' )  ,
```

Comment:
The below code could be integrated in this array. Even if it's static, the configuration options won't change within the current script execution.
Probably a 3rd key is needed, as "select_field", which will be included in the select part. Usually will be the same as the sort_field, but in the user table case, it will contain the alias for the computed column.